### ISSUE TITLE
IBM.Data.Db2 does not get loaded by tracer

### ISSUE THREAD

**Issue by gc-dev1** [DataDog/dd-trace-dotnet#1494]
**Describe the bug**
Using `Net5.IBM.Data.Db2-lnx 5.0.0.200` in basic .NET 5 Web-API to query the database, the DD Tracer doesn't pick up on IDbCommand / DbCommand in the profiler.
No traces happening at database level.

If you need more information or have further questions - just let me know.
Thank you for your help.

**Expected behavior**
Automatically attach to DB integration and trace common db functions.

**Runtime environment:**
- Instrumentation mode: automatic & manual (Nuget package)
- Tracer version: 1.26.2
- OS: Linux, RHEL x64
- CLR: .NET Framework 5.0

**Additional context**
Regardlessly, the automatic tracing picks up everything else works just fine, as expected (HTTP requests, Redis,...)

Trace debug logs attached

[dotnet-tracer-native-dotnet-1052568.log](https://github.com/DataDog/dd-trace-dotnet/files/6549486/dotnet-tracer-native-dotnet-1052568.log)
[dotnet-tracer-managed-dotnet-20210526.log](https://github.com/DataDog/dd-trace-dotnet/files/6549487/dotnet-tracer-managed-dotnet-20210526.log)

**Comment by andrewlock** [comment]
Hi, thanks for raising this and for providing the logs, sorry to hear you're having difficulty with the `DB2Command`. 

I think you're seeing that behaviour because `Net5.IBM.Data.Db2-lnx` targets .NET Standard. We currently don't instrument .NET Standard libraries by default, so that would explain why only the DB2 commands are missing. 

Could you try setting the `DD_TRACE_NETSTANDARD_ENABLED` environment variable to `true`, and testing your application again? That should ensure the `DB2Command` is picked up. Note that you have to set this as an environment variable: setting via _datadog.json_ won't work.

**Comment by gc-dev1** [comment]
Thank you for pointing this out.
I wasn't aware that .NET Standard is actually opt-in. I enabled it and debug log prints out `"netstandard_enabled":true`

After running some tests, the first results are looking promising.
The tracer picks up `Db2Command.executeNonQuery()`
```
2021-05-28 15:09:49.482 +02:00 [DBG] Span closed: [s_id: 7884921900828563704, p_id: 1629569614396696286, t_id: 7510994884111348310] for (Service: some_api-db2, Resource: SELECT COUNT(*) FROM table, Operation: db2.query, Tags: [env (tag):dev,span.kind (tag):client,db.type (tag):db2,component (tag):AdoNet,db.name (tag):ONLINE,out.host (tag):ip:port,])
{ MachineName: ".", Process: "[1575909 dotnet]", AppDomain: "[1 GCO_API]", TracerVersion: "1.26.2.0" }
```

```
05/28/21 03:09:44.411 PM [1575909|1575934] [info] *** JITCompilationStarted() replaced calls from SOME_API.Controllers.WeatherForecastController.testDbQuery() to System.Data.Common.DbCommand.ExecuteNonQuery() 167772383 with calls to Datadog.Trace.ClrProfiler.Integrations.AdoNet.DbCommandIntegration.ExecuteNonQuery() 167772515
```

But it doesn't seem to pick up on `Db2Command.ExecuteReader()`

Example Code
```
var db = new DB2Connection("...");
db.Open();

var cmd = db.CreateCommand();
cmd.CommandText = "SELECT name FROM table";
var reader = cmd.ExecuteReader();

while (reader.Read())
{
    var c = reader.GetString(0);
}

db.Close();
```

Am I missing something?
Or is there an issue with this specific function?

**Comment by andrewlock** [comment]
Glad the initial results were successful! 

The bad news is that it looks like the `Db2Command` implementation [uses `new` to hide the `ExecuteReader`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/new-modifier) implementation in the base `DbCommand`. We instrument the base `DbCommand` implementation, which is never called in their implementation.

One thing you could try is to cast the command to the base type before calling `ExecuteReader()`, and then casting the result back to a `DB2DataReader`:

```csharp
var cmd = db.CreateCommand();
cmd.CommandText = "SELECT name FROM table";
var reader = (DB2DataReader) ((DbCommand)cmd).ExecuteReader();
```

The casting back and forth is a bit cumbersome, but it should resolve your issue!

**Comment by gc-dev1** [comment]
Thanks a lot!
Casting `DbCommand` does work indeed, like you suggested.

But I'm not so sure if this is possible to "tinker-cast" any objects - when we are using `IBM.EntityFrameworkCore-lnx`.
Which we are about to implement in the upcoming weeks.

I guess there aren't that many DB2 users out there. Tracing SQLs is probaly the most relevant key feature for us.
But if it's possible to get .NET Trace to fully support DB2 integration - it would be really helpful and is crucial for us.

**Comment by andrewlock** [comment]
@gc-dev1 in the next version of the tracer (not yet released, after 2.0.1) we will instrument the `DbCommand` base class which should resolve this issue. Do let us know if this isn't the case. Thanks.
